src/Symfony/Component/Security/Core/Encoder/SodiumPasswordEncoder.php

```diff
@@ -80,8 +80,12 @@ public function isPasswordValid(string $encoded, string $raw, ?string $salt): bo
             return false;
         }
 
-        if (72 >= \strlen($raw) && 0 === strpos($encoded, '$2')) {
-            // Accept validating BCrypt passwords for seamless migrations
+        if (0 !== strpos($encoded, '$argon')) {
+            if (72 >= \strlen($raw) && 0 === strpos($encoded, '$2')) {
+                // Accept validating BCrypt passwords for seamless migrations
+                return password_verify($raw, $encoded);
+            }
+
             return password_verify($raw, $encoded);
```

Comment:
a bcrypt hash will reach that call now when the raw password is longer than 72 chars, which won't return false anymore but check the password. This is the wrong logic. It should be changed as such:

```php
if (0 === strpos($encoded, '$2')) {
    return 72 >= \strlen($raw) && password_verify($raw, $encoded);
}

if (0 !== strpos($encoded, '$argon')) {
    return password_verify($raw, $encoded);
}
```